src/frontend/src/flows/recovery/recoverWith/device.ts

```diff
@@ -0,0 +1,71 @@
+import { html, render } from "lit-html";
+import { displayError } from "../../../components/displayError";
+import { unreachable } from "../../../utils/utils";
+import {
+  apiResultToLoginFlowResult,
+  LoginFlowSuccess,
+} from "../../login/flowResult";
+import { DeviceData } from "../../../../generated/internet_identity_types";
+import { IIConnection } from "../../../utils/iiConnection";
+
+const pageContent = (userNumber: bigint) => html`
+  <style>
+    .full-width {
+      width: 100%;
+    }
+  </style>
+  <div class="container full-width">
+    <h1>Recovery for ${userNumber}</h1>
+    <p>
+      You are about to recover your anchor using a recovery device. Please click
+      "continue" and then follow your browser's instructions to connect your
+      device.
+    </p>
+    <button id="recover-with-device__continue" class="primary">Continue</button>
+    <button id="recover-with-device__cancel">Cancel</button>
+  </div>
+`;
+
+export const deviceRecoveryPage = async (
+  userNumber: bigint,
+  device: DeviceData
+): Promise<LoginFlowSuccess | null> => {
+  const container = document.getElementById("pageContent") as HTMLElement;
+  render(pageContent(userNumber), container);
+  return init(userNumber, device);
+};
+
+const init = (
+  userNumber: bigint,
+  device: DeviceData
+): Promise<LoginFlowSuccess | null> =>
+  new Promise((resolve) => {
+    const buttonContinue = document.getElementById(
+      "recover-with-device__continue"
+    ) as HTMLButtonElement;
+    buttonContinue.onclick = async () => {
+      const result = apiResultToLoginFlowResult(
+        await IIConnection.fromWebauthnDevices(userNumber, [device])
+      );
+
+      switch (result.tag) {
```

Comment:
so the promise just never resolves if errors happens (not familiar with error mgmt in ii)?